src/Component/Tests/Dummy/ClassAsParameterGrid.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Grid\Tests\Dummy;
+
+use Sylius\Bundle\GridBundle\Builder\GridBuilderInterface;
+use Sylius\Bundle\GridBundle\Grid\AbstractGrid;
+use Sylius\Bundle\GridBundle\Grid\ResourceAwareGridInterface;
+
+final class ClassAsParameterGrid extends AbstractGrid implements ResourceAwareGridInterface
+{
+    private string $authorClass;
+
+    public function __construct(string $authorClass)
+    {
+        $this->authorClass = $authorClass;
+    }
+
+    public static function getName(): string
+    {
+        return 'app_class_as_parameter';
+    }
+
+    public function getResourceClass(): string
+    {
+        return $this->authorClass;
```

Comment:
only better it could work, if we would be able to use parameter directly here. But really neat improvement

Reply:
Yes, but I don't know how...